kano_settings/system/advanced.py

```diff
@@ -282,6 +285,10 @@ def set_hosts_blacklist(enable, block_search,
     if enable:
         logger.debug('enabling blacklist')
 
+        if os.path.exists(hosts_file_backup):
+            logger.debug('restoring original backup file')
+            shutil.copy(hosts_file_backup, hosts_file)
```

Comment:
If we do this and the backup file is large then we end up copying this file and then copying it back again. Also here we do `shutil.copy()` and a few lines down we do `shutil.copyfile`; is there a difference? Should one be preferred?